pennylane/ops/op_math/exp.py

```diff
@@ -0,0 +1,240 @@
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines the symbolic operation that stands for an operator raised to a power.
+"""
+from warnings import warn
+from scipy.sparse.linalg import expm as sparse_expm
+
+import pennylane as qml
+from pennylane import math
+from pennylane.operation import (
+    expand_matrix,
+    Tensor,
+    OperatorPropertyUndefined,
+)
+from pennylane.wires import Wires
+
+from .symbolicop import SymbolicOp
+
+
+def exp(op, coeff=1, id=None):
+    """Take the exponential of an Operator times a coefficient.
+
+    Args:
+        base (~.operation.Operator): The Operator to be exponentiated
+        coeff=1 (Number): A scalar coefficient of the operator.
+
+    Returns:
+       :class:`Exp`: A :class`~.operation.Operator` representing an operator exponential.
+
+    **Example**
+
+    This symbolic operator can be used to make general rotation operators:
+
+    >>> x = np.array(1.23)
+    >>> op = qml.exp( qml.PauliX(0), -0.5j * x)
+    >>> qml.math.allclose(op.matrix(), qml.RX(x, wires=0).matrix())
+    True
+
+    This can even be used for more complicated generators:
+
+    >>> t = qml.PauliX(0) @ qml.PauliX(1) + qml.PauliY(0) @ qml.PauliY(1)
+    >>> isingxy = qml.exp(t, 0.25j * x)
+    >>> qml.math.allclose(isingxy.matrix(), qml.IsingXY(x, wires=(0,1)).matrix())
+    True
+
+    If the coefficient is purely imaginary and the base operator is Hermitian, then
+    the gate can be used in a circuit, though it may not be supported by the device and
+    may not be differentiable.
+
+    >>> @qml.qnode(qml.device('default.qubit', wires=1))
+    ... def circuit(x):
+    ...     qml.exp(qml.PauliX(0), -0.5j * x)
+    ...     return qml.expval(qml.PauliZ(0))
+    >>> print(qml.draw(circuit)(1.23))
+    0: ──Exp─┤  <Z>
+
+    If the base operator is Hermitian and the coefficient is real, then the ``Exp`` operator
+    can be measured as an observable:
+
+    >>> obs = qml.exp(qml.PauliZ(0), 3)
+    >>> @qml.qnode(qml.device('default.qubit', wires=1))
+    ... def circuit():
+    ...     return qml.expval(obs)
+    >>> circuit()
+    tensor(20.08553692, requires_grad=True)
+
+    """
+    return Exp(op, coeff, id=id)
+
+
+class Exp(SymbolicOp):
+    """A symbolic operator representating the exponential of a operator.
+
+    Args:
+        base (~.operation.Operator): The Operator to be exponentiated
+        coeff=1 (Number): A scalar coefficient of the operator.
```

Comment:
Is this normally how we display default values of kwargs in PL docs? I've seen it under the separate heading `Kwargs:` and other times just listed in the description of the args.

Reply:
I tend to think of the difference between Args and Kwargs in terms of their frequency of use.  I think `coeff` will be used frequently enough that it should be an "Arg".